src/global/signal_handler.cc

```diff
@@ -249,6 +292,7 @@ struct SignalHandler : public Thread {
 	  if (handlers[signum]) {
 	    r = read(handlers[signum]->pipefd[0], &v, 1);
 	    if (r == 1) {
+             handlers[signum]->siginfo_handler(signum, &handlers[signum]->info_t, (void *)NULL);
```

Comment:
This is pointless--we always set it to handle_global_signal_info and the user can't/doesn't customize it.  Get rid of that function, and just inline the derr line right here to log the info.